app/models/account_recovery_request.rb

```diff
@@ -0,0 +1,14 @@
+class AccountRecoveryRequest < ApplicationRecord
+  belongs_to :candidate
+  belongs_to :previous_candidate, optional: true, class_name: 'Candidate'
+
+  validates :code, presence: true
+
+  normalizes :previous_account_email, with: ->(email) { email.downcase.strip }
+
+  def self.generate_code
+    code = SecureRandom.random_number(100_000..999_999)
```

Comment:
This means we wouldn't have any codes that start with 0. If we want 6 random digits, rather than a number between `100_000` and `999_999`, we should do something like `6.times.collect { rand(0...9) }.join
` instead -- and it would need to be a string rather that an integer if we want to persist the leading zeros. (also, in general, things with digits aren't numbers unless we are going to do maths to them, eg phone numbers and credit card numbers are strings rather than integers)

Reply:
Yeah, I agree. I'll change it to a true random 6 digit string as you suggested, thanks 👍🏻